Anchorage.swift

```diff
@@ -289,255 +317,239 @@ public struct LayoutExpression<T : LayoutAnchorType> {
-                    let expression = (otherX + edge.transform(constant: c)) ~ priority
-                    return builder.horizontalBuilder(forEdge: edge)(x, expression)
-
-                case let (y as NSLayoutYAxisAnchor, otherY as NSLayoutYAxisAnchor):
-                    let expression = (otherY + edge.transform(constant: c)) ~ priority
-                    return builder.verticalBuilder(forEdge: edge)(y, expression)
-
-                default:
-                    preconditionFailure("Layout axis of constrained anchors must match.")
-                }
-            }()
-        }
-
-        return edgeConstraints
+        let horizontalConstraints = horizontalAnchors.constraints(forAnchors: anchors.horizontalAnchors, constant: c, priority: priority, builder: builder)
+        let verticalConstraints = verticalAnchors.constraints(forAnchors: anchors.verticalAnchors, constant: c, priority: priority, builder: builder)
+        return ConstraintGroup(top: verticalConstraints.top,
+                               leading: horizontalConstraints.leading,
+                               bottom: verticalConstraints.bottom,
+                               trailing: verticalConstraints.trailing,
+                               centerX: horizontalConstraints.centerX,
+                               centerY: verticalConstraints.centerY)
     }
 
 }
 
-// MARK: - EdgeConstraints
+// MARK: - ConstraintGroup
 
-public struct EdgeConstraints {
+public struct ConstraintGroup {
 
     public var top: NSLayoutConstraint?
     public var leading: NSLayoutConstraint?
     public var bottom: NSLayoutConstraint?
     public var trailing: NSLayoutConstraint?
+    public var centerX: NSLayoutConstraint?
+    public var centerY: NSLayoutConstraint?
 
     public var horizontal: [NSLayoutConstraint] {
-        return [leading, trailing].flatMap { $0 }
+        return [leading, trailing, centerX].flatMap { $0 }
     }
 
     public var vertical: [NSLayoutConstraint] {
-        return [top, bottom].flatMap { $0 }
+        return [top, bottom, centerY].flatMap { $0 }
     }
 
     public var all: [NSLayoutConstraint] {
-        return [top, leading, bottom, trailing].flatMap { $0 }
-    }
-
-    subscript (edge: LayoutEdge) -> NSLayoutConstraint? {
-        get {
-            switch edge {
-            case .top:      return top
-            case .leading:  return leading
-            case .bottom:   return bottom
-            case .trailing: return trailing
-            }
-        }
-
-        set {
-            switch edge {
-            case .top:      top = newValue
-            case .leading:  leading = newValue
-            case .bottom:   bottom = newValue
-            case .trailing: trailing = newValue
-            }
-        }
+        return [top, leading, bottom, trailing, centerX, centerY].flatMap { $0 }
     }
 
 }
 
 // MARK: - Constraint Builders
 
-private struct ConstraintBuilder {
+struct ConstraintBuilder {
```

Comment:
Yeah I'm not sure if it really matters since these aren't marked as `public` and therefore can't be accessed outside the module anyway. If someone drags in the `.swift`, they'll have access outside the file, but most people will probably incorporate via Cocoapods